Animation for locked driver door

rt-2 opened the pull request:
I figured nobody had time to do this.
As I was watching the code, I felt like it was almost done just because I was looking at it, felt easy.
I though that copying the script would be easy, but I had to understand it completely to document it, well now it's done.
Side note: line 1220 was missing the _2

rt-2 committed:
Animation for locked driver door

I figured nobody had time to do this.
As I was watching the code, I felt like it was almost done just because I was looking at it, felt easy.
I though that copying the script would be easy, but I had to understand it completely to document it, well now it's done.
Side note: line 1220 was missing the _2

rt-2 committed:
Update README.md

I though that copying the script would be easy, but I had to understand it completely to document it, well now it's done.
Thanks to ziggi.

ziggi commented on fixes.inc:
```diff
@@ -9398,6 +9515,78 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+native BAD_GetVehicleParamsEx(vehicleid, &engine, &lights, &alarm, &doors, &bonnet, &boot, &objective) = GetVehicleParamsEx;
+
+#if FIX_OnPlayerEnterVehicle_3
+	stock FIXES_GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective)
```
You forgot about ampersands.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/5ce6d8ae3e6b74c47d74908d1ecefd47c8d13288

ziggi commented on fixes.inc:
```diff
@@ -9398,6 +9515,78 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+#if FIX_OnPlayerEnterVehicle_3
+	stock FIXES_GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective)
+	{
+		return GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective);
```
I think what `return` should be replaced by variable and moved to the end of the function.
```Pawn
new
    ret = GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective);
//...
return ret;
```

rt-2 replied:
Yes totally, thank you that didn't work.
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/8e91421fb316098e8ca262412ec54b4cb9c49d0c

ziggi commented on fixes.inc:
```diff
@@ -9398,6 +9515,78 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+ */
+#if !FIXES_Single
+	#if FIX_OnPlayerEnterVehicle_3
+		#if !defined FILTERSCRIPT
```
This can be written in one line:
```pawn
#if FIX_OnPlayerEnterVehicle_3 && !FIXES_Single && !defined FILTERSCRIPT
```

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/ebf57a9e5d1187e02a51bd3cef42b17245111925

ziggi commented on fixes.inc:
```diff
@@ -6406,6 +6498,31 @@ native BAD_TogglePlayerControllable(playerid, toggle) = TogglePlayerControllable
+ */
+
+#if FIX_OnPlayerEnterVehicle_3
+	static stock FIXES_StopPlayerEnterVehicle(playerid)
```
The function should be renamed to `_FIXES_StopPlayerEnterVehicle` ([guidelines](https://github.com/Open-GTO/sa-mp-fixes/blob/master/fixes.inc#L859-L861)).

rt-2 replied:
Damn and I saw that when I was going through the file and forgot it then!
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/4cdd88d0132a87f3cc6f99fbae52f5df73ae5b88

ziggi commented on fixes.inc:
```diff
@@ -6406,6 +6498,31 @@ native BAD_TogglePlayerControllable(playerid, toggle) = TogglePlayerControllable
+	static stock FIXES_StopPlayerEnterVehicle(playerid)
+	{
+		player_data[playerid][e_IsTrying] = false;
+		player_data[playerid][e_IsOpening] = false;
```
Wrong variables.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/3a5cf9bc8d61f0c38c53e9ae67936fe686832d48

ziggi commented on fixes.inc:
```diff
@@ -2233,15 +2249,51 @@ static stock
+	 * to enter a locked vehicle and the animation
+	 */
+#if FIX_OnPlayerEnterVehicle_3
+	bool:FIXES_gsPlayerVehEntering[MAX_PLAYERS],
```
We have `FIXES_gsPlayerBools` array for booleans.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/e313629d2a3808463ba7ab903dadc182e831d41c

ziggi commented on fixes.inc:
```diff
@@ -2233,15 +2249,51 @@ static stock
+	 * Stores the locked vehicle's id that the player
+	 * is currently trying to enter
+	 */
+	FIXES_gsPlayerVehID[MAX_PLAYERS],
```
I think that `FIXES_gsPlayerVehicleID` name is better.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/13ddbf630b6660106602b30c8fa980d33dcda878

ziggi commented on fixes.inc:
```diff
@@ -2233,15 +2249,51 @@ static stock
+	 * Stores the server tick when the player started
+	 * trying to enter the locked vehicle
+	 */
+	FIXES_gsPlayerVehTickCount[MAX_PLAYERS],
```
Same, `FIXES_gsPlayerVehicleTickCount`.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/13ddbf630b6660106602b30c8fa980d33dcda878

ziggi commented on fixes.inc:
```diff
@@ -2233,15 +2249,51 @@ static stock
+	 * Boolean: true if the player is currently doing
+	 * the animation
+	 */
+	bool:FIXES_gsPlayerVehTrying[MAX_PLAYERS],
```
Same about booleans.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/e313629d2a3808463ba7ab903dadc182e831d41c

ziggi commented on fixes.inc:
```diff
@@ -6308,6 +6382,24 @@ native BAD_TogglePlayerControllable(playerid, toggle) = TogglePlayerControllable
+		//  BEGIN: OnPlayerEnterVehicle_3
+		// ===============================
+		#if FIX_OnPlayerEnterVehicle_3
+			if (FIXES_gsVehicleIsLocked[vehicleid]) {
```
In fixes.inc we move the bracket on the next line.

rt-2 replied:
Yeah sorry I do that too in my project, I think it's more visible.
I copied this line from a sloppy programmer https://github.com/ziggi/samp-enterfix/blob/master/enterfix.inc#L34 sorry xD!
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/e12925e4f15cfd10bf2081e1cd605584dae09157

ziggi replied:
I prefer this code style, but fixes.inc have different style.

ziggi commented on fixes.inc:
```diff
@@ -205,6 +205,15 @@
+ *           to enter a 'locked' vehicle.
+ * SEE:      "OnPlayerEnterVehicle".
+ * AUTHOR:   ziggi (https://github.com/ziggi)
+ * AUTHOR:   rt-2 (https://github.com/Y-Less/)
```
Wrong url.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/971ae18732dedfa888b40df0596e079bf731605c

ziggi commented on fixes.inc:
```diff
@@ -205,6 +205,15 @@
+ * SEE:      "OnPlayerEnterVehicle".
+ * AUTHOR:   ziggi (https://github.com/ziggi)
+ * AUTHOR:   rt-2 (https://github.com/Y-Less/)
+ * POST:     https://github.com/ziggi/samp-enterfix
```
Original post about this bug here: http://forum.sa-mp.com/showthread.php?t=560019

IstuntmanI replied:
Confirmed: rt-2 is Y-Less undercover.

rt-2 replied:
I wish I was Y_Less, basically the equivalent of Jesus to the SAMP Forum.
Anyhow I have still been uncovered: I copy my stuff from Y_Less works haha.

rt-2 replied:
Should I add the forum post or replace by the forum post?

rt-2 replied:
I decided to keep only the forum post. No where else there was a github link as the original post, and the repo link is in ziggi's profile which is credited, and all other fixes had only one 'post'.
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/c7a127cc66ca7fd9065c86092ebdf31cad02192c

ziggi replied:
Yes, of course.

ziggi commented on fixes.inc:
```diff
@@ -2233,15 +2249,51 @@ static stock
 	 *
 	 * A player's IP as a 32-bit integer.
 	 */
-#if FIXES_SilentKick && (FIX_OnVehicleMod || FIX_OnPlayerEnterVehicle || FIX_OnPlayerEnterVehicle_2)
```
What the point of this changes?

rt-2 replied:
This must be a mistake, because I didn't removed the closing of the if anyway

rt-2 replied:
Now I see better, this have no effect at all so I replaced it to its original position.
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/9c113e0d4dec89173612e3ce353a7f239f01c6b6

ziggi reviewed (changes requested):
Thanks, but some improvements are required.

IstuntmanI commented:
That's exactly why I don't make pull requests that are too big ! XD

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134015180

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134015789

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014085

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134011839

rt-2 committed:
Update fixes.inc

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134012702

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134013787

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014972

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014333 and https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014439

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014146 and https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134014498

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134015533

rt-2 commented:
Well that should do it, let me know if you have anything else?
Thank you,
rt-2

rt-2 reviewed (commented):
First time doing this, am I at the right spot?

ziggi commented on fixes.inc:
```diff
@@ -5758,6 +5799,26 @@ native BAD_GetWeaponName(weaponid, weapon[], len) = GetWeaponName;
+		//  BEGIN: OnPlayerEnterVehicle_3
+		// =============================
+		#if FIX_OnPlayerEnterVehicle_3
+			if (FIXES_gsPlayerBools[playerid] & e_FIXES_BOOLS_VEH_ENTER) {
```
Same about brackets :).

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/b3ab260be1cf5989508d6c6a4ca6e3ed9fc8a33c and https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/30cb65981ed54208a3474a50c8646b5e4158816d

ziggi commented on fixes.inc:
```diff
@@ -6406,6 +6487,32 @@ native BAD_TogglePlayerControllable(playerid, toggle) = TogglePlayerControllable
+		FIXES_gsPlayerBools[playerid] &= ~e_FIXES_BOOLS_VEH_ENTER;
+		FIXES_gsPlayerBools[playerid] &= ~e_FIXES_BOOLS_VEH_ENTER_ANIM;
+
+		new vehicleid = FIXES_gsPlayerVehicleID[playerid],
```
Move to the next line:
```Pawn
new
	vehicleid = FIXES_gsPlayerVehicleID[playerid],
	engine, lights, alarm, doors, bonnet, boot, objective;
```

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/ebd54742738cce835c6c9f47dd63fad01263cd96

ziggi commented on fixes.inc:
```diff
@@ -6406,6 +6487,32 @@ native BAD_TogglePlayerControllable(playerid, toggle) = TogglePlayerControllable
+#if FIX_OnPlayerEnterVehicle_3
+	static stock _FIXES_StopPlayerEnterVehicle(playerid)
+	{
+	
```
Extra line.

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/018e84deae22734592d70ff4b19cf12485fa3d45

ziggi commented on fixes.inc:
```diff
@@ -9398,6 +9505,76 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+ */
+#if FIX_OnPlayerEnterVehicle_3 && !FIXES_Single && !defined FILTERSCRIPT
+	forward FIXES_GetVehicleDoorsEx(vehicleid);
+	public FIXES_GetVehicleDoorsEx(vehicleid)
```
This should be also with _ prefix. And add one line between.
```Pawn
	forward _FIXES_GetVehicleDoorsEx(vehicleid);

 	public _FIXES_GetVehicleDoorsEx(vehicleid)
```

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/a9859fd79ed82114192afb22510bda4dc497d43d

ziggi commented on README.md:
```diff
@@ -65,6 +65,7 @@ TextDrawSetString | Crashes on a blank string and size greater than 1024. | TomT
 AllowInteriorWeapons | Does nothing. | KoczkaHUN |  |
 OnPlayerEnterVehicle | Crashes other players when people enter an invalid seat. | RyDeR`/[Y_Less](https://github.com/Y-Less/) | [Link](http://forum.sa-mp.com/showpost.php?p=1410296) |
 OnPlayerEnterVehicle_2 | Crashes the server when hacks enter an invalid vehicle. | im/[Y_Less](https://github.com/Y-Less/) |  |
+OnPlayerEnterVehicle_3 | No player animation when trying to enter the driver door of a locked vehicle. | [ziggi](https://github.com/ziggi)/[rt-2](https://github.com/rt-2) | [Link](https://github.com/ziggi/samp-enterfix) |
```
Wrong post url, should be: http://forum.sa-mp.com/showthread.php?t=560019

rt-2 replied:
https://github.com/Open-GTO/sa-mp-fixes/pull/64/commits/cb41de179daa0e209d72f50d68d08543bc61c3b6

ziggi reviewed (changes requested):
Almost fine.

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134082892

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134082839

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134082861

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134082824

rt-2 committed:
Update fixes.inc

https://github.com/Open-GTO/sa-mp-fixes/pull/64#discussion_r134082824

rt-2 reviewed (commented):
Thank you

rt-2 committed:
Update README.md

Y-Less commented:
Honestly, I'm not even bothered about things like how many blank lines there are between `forward` and `public`, or if certain functions start with `_`.  I generally use that prefix when I can't make functions `static` (so publics), but need to indicate that they are internal and shouldn't really be called by users.  That's the only reason it is that way.

In fact, often I wouldn't even complain about brace position in a code review.  Consistency is nice, but they should be mosty used for actual logical problems IMHO.  Unless of course the code is truly ugly.

Y-Less commented:
One thing I will comment on though:

`(` - that is a bracket.

`{` - that is a brace.

They are different in both use and name.  For completion:

`<` - angle bracket.

`[` - square bracket.

rt-2 commented:
I think it's nice to have uniformity in the code it's easier to read after. Especially in this case where a lot  of people working on the same file.
